### ISSUE TITLE
Unable to set spot_price to default in aws_launch_configuration

### ISSUE THREAD

**Issue by olgafandeliuk** [hashicorp/terraform-provider-aws#12206]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.14
provider.aws: version = "~> 2.50"

### Affected Resource(s)

aws_launch_configuration

### Terraform Configuration Files

```
resource "aws_launch_configuration" "some_name" {
  count = "${var.use_spot == "yes" ? 1 : 0}"

  name_prefix          = "${var.name}-"
  image_id             = "${var.id}"
  ebs_optimized        = "${var.ebs_optimized}"
  instance_type        = "${var.instance_type}"
  spot_price           = "${var.spot_price}"
  iam_instance_profile = "${var.instance_profile_name}"
  key_name             = "${var.key_name}"
  security_groups      = ["${var.security_group_ids}"]

  user_data = "${var.user_data == "" ?
                            data.template_file.userdata.rendered :
                            var.user_data}"

  lifecycle {
    create_before_destroy = true
  }
}
```
in variables.tf set:
```spot_price="" ```

### Expected Behavior

If I set spot_price="", I expect to use spot instances with maximum spot_price equal to the On-Demand price

### Actual Behavior

EC2 instances from launch configuration, where spot_price="" are on-demand instead of being spot with default price set to on-demand price
https://aws.amazon.com/blogs/compute/new-amazon-ec2-spot-pricing/ 

### Steps to Reproduce

1. Set spot_price="" in launch_configuration, 
2. Use this launch_configuration in autoscaling group
3. Bring up ec2 instance from autoscaling group
4. aws ec2 describe-instances --instance-ids $instance-id and verify, that there is no "SpotInstanceRequestId" or  "InstanceLifecycle", because instance is on-demand instead of spot

### References

https://www.terraform.io/docs/providers/aws/r/launch_configuration.html
according to the docs:
spot_price - (Optional; Default: On-demand price) The maximum price to use for reserving spot instances.

**Comment by KIVagant** [comment]
I would be nice to have a separate option that allows to enable the "spot" EC2 lifecycle without setting the price.

**Comment by JamiKarvanen** [comment]
@justinretzolk This still doesn't seem to work as it should. If I set spot_price to "", it creates an on-demand ASG instead :-/

**Comment by justinretzolk** [comment]
Hey @JamiKarvanen, thanks for following up. I took a look over the documentation for the resource and followed the link that it has to information on spot instances ([here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-spot-instances.html), admittedly, spot instances are not my area of expertise). In reviewing that documentation, I'm wondering if this may be a matter of how those are defined rather than a bug in the provider. From that linked doc:

> A Spot Instance is an instance that uses spare EC2 capacity that is available for less than the On-Demand price.

Based on this definition, I would expect that spot instances having a price equal to the on-demand price would be incompatible with the idea of spot instances; does that read feel correct to you?

**Comment by kahlil29** [comment]
@justinretzolk 
Picking this up since I just got hit by this issue. 

It is common practice among companies that use AWS EC2 Spot Instances to set the `spot_price` at the `On-Demand` price. It's like preparing for a scenario that we never expect to happen. 

Usually, people/companies that use spot instances aggressively and intensively, put their `spot_price` at values that are relatively much closer to the current `spot_price` set by AWS. The behaviour that they want is that if the `spot_price` for running a Spot Instance goes above their price, they are okay with releasing the capacity/instance rather than paying for it. 

In my (and I'm sure this is the case with a lot of other companies/people) case, I want to specify my `spot_price` as the `On-Demand` price because I don't care if the `spot_price` goes up by 10-30% or so (rare that it goes up so much), since I am still saving $$ by paying much less than I would for an on-demand instance. 

I hope I've explained the above correctly. 
Also, if we look at the documentation for the newer `launch_template` https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#spot_max_price 
I am currently using this with Terraform/AWS and it works without an issue. 

Could we have the same behaviour for the `aws_launch_configuration` -> `spot_price` attribute as well?
